### PR TITLE
chore: update Gradle configuration

### PR DESCRIPTION
**Pull request by TheOneWithTheBraid**
- update Gradle wrapper
- remove accidentally committed JAR blob of Gradle wrapper
- bump Kotlin Gradle plugin
- bump Android Gradle plugin
- migrate JVM toolchain declaration to android level
- declare Kotlin compatibility API version

### CODE REVIEW

File: connector/build.gradle [hunk 1]
```diff
@@ -6,12 +6,20 @@ plugins {
...
+
+    compileOptions{
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+
+    kotlinOptions {
+        jvmTarget = '1.8'
+        apiVersion = '1.7'
+        languageVersion = '2.0'
```
Comment (p1gp1g) [inline connector/build.gradle:20]: Shouldn't we set languageVersion only to 1.7 ?
And it is better to use KotlinVersion.KOTLIN_1_7, so we get warned when deprecated

```suggestion
        # languageVersion provides source compatibility with the specified version of Kotlin
        languageVersion = KotlinVersion.KOTLIN_1_7
```

Comment (TheOneWithTheBraid) [reply connector/build.gradle:20]: Corrected it. Though the KotlinVersion enum would require us to import JetBrains Kotlin DSL.

Comment (p1gp1g) [reply connector/build.gradle]: Ok, that's fine with the string then

**Comment by p1gp1g** [comment]
Thank you for your contribution !

**Pull request merged**